assets/js/googlesitekit/data/create-fetch-store.test.js

```diff
@@ -0,0 +1,328 @@
+				} );
+			} );
+
+			it( 'yields the expected actions for an error request', () => {
+				const fetchStoreDefinition = createFetchStore( {
+					baseName: 'SaveSomeData',
+					controlCallback: async () => true,
+					reducerCallback: ( state ) => state,
+				} );
+
+				const action = fetchStoreDefinition.actions.fetchSaveSomeData();
+
+				const error = {
+					code: 'this-went-wrong',
+					message: 'This went wrong.',
+				};
+
+				expect( action.next().value.type ).toEqual( 'START_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value.type ).toEqual( 'FETCH_SAVE_SOME_DATA' );
+				expect( action.throw( error ).value.type ).toEqual( 'CATCH_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value ).toEqual( {
+					response: undefined,
+					error,
+				} );
+			} );
+
+			it( 'makes a network request based on controlCallback', async () => {
+				const expectedResponse = 'response-value';
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/test\/data\/some-data/
+					)
+					.mockResponseOnce(
+						JSON.stringify( expectedResponse ),
+						{ status: 200 }
+					);
+
+				const { response, error } = await dispatch.fetchGetSomeData( {}, 'value-to-key-response-by' );
+
+				expect( error ).toEqual( undefined );
+				expect( response ).toEqual( expectedResponse );
+				expect( store.getState().data ).toEqual( {
+					'value-to-key-response-by': expectedResponse,
+				} );
+			} );
+
+			it( 'dispatches an error if the request fails', async () => {
+				const errorResponse = {
+					code: 'internal_server_error',
+					message: 'Internal server error',
+					data: { status: 500 },
+				};
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/test\/data\/some-data/
+					)
+					.mockResponseOnce(
+						JSON.stringify( errorResponse ),
+						{ status: 500 }
+					);
+
+				muteConsole( 'error' );
+				const { response, error } = await dispatch.fetchGetSomeData( {}, 'value-to-key-response-by' );
+
+				expect( error ).toEqual( errorResponse );
+				expect( response ).toEqual( undefined );
+				expect( store.getState().data ).toEqual( undefined );
+			} );
+
+			it( 'sets flag for request being in progress', async () => {
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/test\/data\/some-data/
+					)
+					.mockResponseOnce(
+						JSON.stringify( { someValue: 42 } ),
+						{ status: 200 }
+					);
+
+				const requestArgs = [ {}, 'aValue' ];
+
+				// Initially the request is not in progress..
+				expect( select.isFetchingGetSomeData( ...requestArgs ) ).toEqual( false );
+
+				dispatch.fetchGetSomeData( ...requestArgs );
+
+				// Now it should be in progress.
+				expect( select.isFetchingGetSomeData( ...requestArgs ) ).toEqual( true );
+
+				// A request for other arguments however is not in progress.
+				expect( select.isFetchingGetSomeData( {}, 'anotherValue' ) ).toEqual( false );
+
+				await subscribeUntil( registry,
+					() => store.getState().data !== undefined,
```

Comment:
Let's wait until the selector `hasFinishedResolution` - I think we should generally await selectors this way unless the test requires otherwise for some reason (I can't think of why that would be necessary though).

Reply:
There's no resolver involved here that we can rely on, and we shouldn't rely on the `isFetchingGetSomeData` selector since that's what's actually being tested below.

Since we're really just testing a plain fetch store here, all that we can do is inspect the store state directly to know that the reducer has completed setting the data as expected.

Reply:
Oh right - the resolver would be defined outside of this along with its selector 👍